code/modules/integrated_electronics/core/assemblies.dm

```diff
@@ -0,0 +1,372 @@
+			HTML += "<a href=?src=[REF(circuit)];bottom=[REF(circuit)];from_assembly=1>\[Move to Bottom\]</a>"
+			HTML += "<br>"
+
+	HTML += "</body></html>"
+	user << browse(jointext(HTML,null), "window=assembly-\[REF(src)];size=600x350;border=1;can_resize=1;can_close=1;can_minimize=1")
+
+/obj/item/device/electronic_assembly/Topic(href, href_list[])
+	if(..())
+		return 1
+
+	if(href_list["rename"])
+		rename(usr)
+
+	if(href_list["remove_cell"])
+		if(!battery)
+			to_chat(usr, "<span class='warning'>There's no power cell to remove from \the [src].</span>")
+		else
+			var/turf/T = get_turf(src)
+			battery.forceMove(T)
+			playsound(T, 'sound/items/Crowbar.ogg', 50, 1)
+			to_chat(usr, "<span class='notice'>You pull \the [battery] out of \the [src]'s power supplier.</span>")
+			battery = null
+
+	interact(usr) // To refresh the UI.
+
+/obj/item/device/electronic_assembly/verb/rename()
+	set name = "Rename Circuit"
+	set category = "Object"
+	set desc = "Rename your circuit, useful to stay organized."
+
+	var/mob/M = usr
+	if(!check_interactivity(M))
+		return
+
+	var/input = reject_bad_name(input("What do you want to name this?", "Rename", src.name) as null|text,1)
+	if(src && input)
+		to_chat(M, "<span class='notice'>The machine now has a label reading '[input]'.</span>")
+		name = input
+
+/obj/item/device/electronic_assembly/proc/can_move()
+	return FALSE
+
+/obj/item/device/electronic_assembly/drone/can_move()
+	return TRUE
+
+/obj/item/device/electronic_assembly/update_icon()
+	if(opened)
+		icon_state = initial(icon_state) + "-open"
+	else
+		icon_state = initial(icon_state)
+/*
+/obj/item/device/electronic_assembly/GetAccess()
+	. = list()
+	for(var/obj/item/integrated_circuit/part in contents)
+		. |= part.GetAccess()
+
+/obj/item/device/electronic_assembly/GetIdCard()
+	. = list()
+	for(var/obj/item/integrated_circuit/part in contents)
+		var/id_card = part.GetIdCard()
+		if(id_card)
+			return id_card
+*/
+/obj/item/device/electronic_assembly/examine(mob/user)
+
+	for(var/obj/item/integrated_circuit/IC in contents)
+		IC.external_examine(user)
+	for(var/obj/item/integrated_circuit/output/screen/S in contents)
+		if(S.stuff_to_display)
+			to_chat(user, "There's a little screen labeled '[S]', which displays '[S.stuff_to_display]'.")
+		if(opened)
+			interact(user)
+
+/obj/item/device/electronic_assembly/proc/get_part_complexity()
+	. = 0
+	for(var/obj/item/integrated_circuit/part in contents)
+		. += part.complexity
+
+/obj/item/device/electronic_assembly/proc/get_part_size()
+	. = 0
+	for(var/obj/item/integrated_circuit/part in contents)
+		. += part.size
+
+// Returns true if the circuit made it inside.
+/obj/item/device/electronic_assembly/proc/add_circuit(var/obj/item/integrated_circuit/IC, var/mob/user)
+	if(!opened)
+		to_chat(user, "<span class='warning'>\The [src] isn't opened, so you can't put anything inside.  Try using a crowbar.</span>")
+		return FALSE
+
+	if(IC.w_class > src.w_class)
```

Comment:
Drop the src.

Reply:
What src I need to drop? It is needed there in text.

Reply:
src.w_class, If there's no name clashes between local variables and properties adding src is just needless overhead.